File: go/problem_02.go
package main

import "fmt"

func main() {
	fibSequence := []int{0, 1}
	lastNumber := 1
	i := 2
	for {
		lastNumber = fibSequence[i-1] + fibSequence[i-2]
		if lastNumber > 4000000 {
			break
		}
		fibSequence = append(fibSequence, lastNumber)
		i++
	}
	sum := 0
	for _, x := range fibSequence {
		if x%2 == 0 {
			sum += x
		}
	}
	fmt.Println(sum)
}

File: php/problem_02.php
<?php
$fib_sequence = [0, 1];
$last_number = 1;
$i = 2;
while (true) {
    $last_number = $fib_sequence[$i-1] + $fib_sequence[$i-2];
    if ($last_number > 4000000) {
        break;
    }
    $fib_sequence[] = $last_number;
    $i++;
}
echo array_sum(array_filter($fib_sequence, static function($x) {
    return $x % 2 === 0;
}));